Library/Formula/llvm.rb

```diff
@@ -116,18 +151,74 @@ def install
     # Apple's libstdc++ is too old to build LLVM
     ENV.libcxx if ENV.compiler == :clang
 
-    if build.with?("lldb") && build.without?("clang")
-      raise "Building LLDB needs Clang support library."
+    if build.with? "clang"
+      (buildpath/"tools/clang").install resource("clang")
     end
 
-    if build.with? "clang"
+    if build.with? "extra-tools"
+      if build.with? "clang"
+        (buildpath/"tools/clang/tools/extra").install resource("extra-tools")
+      else
+        odie "clang's extra tools must be built with clang"
+      end
+    end
+
+    if build.with? "libcxx"
       (buildpath/"projects/libcxx").install resource("libcxx")
-      (buildpath/"tools/clang").install resource("clang")
-      (buildpath/"tools/clang/tools/extra").install resource("clang-tools-extra")
+      (buildpath/"projects/libcxxabi").install resource("libcxxabi")
+    end
+
+    if build.with? "lld"
+      (buildpath/"tools/lld").install resource("lld")
+    end
```

Comment:
I think I used the nesting to avoid going over 80 characters. If over 80 characters still fits comfortably, I can change that.